records_mover/creds/base_creds.py

```diff
@@ -129,20 +133,51 @@ def default_gcs_client(self) -> Optional['google.cloud.storage.Client']:
             return self.__default_gcs_client
         try:
             import google.cloud.storage  # noqa
-        except ModuleNotFoundError:
-            logger.debug("google.cloud.storage not installed",
-                         exc_info=True)
+        except ModuleNotFoundError as e:
+            logger.warning(f"google.cloud.storage not installed: {e}")
+            logger.debug("Details:", exc_info=True)
             self.__default_gcs_client = None
             return self.__default_gcs_client
 
         try:
-            self.__default_gcs_client = google.cloud.storage.Client(credentials=gcs_creds)
+            other_args = {}
+            #
+            # For mvrec especially, Records Mover on the command line
+            # tends to use "application default creds", which act like
+            # service user creds but are retrieved by browser oauth.
+            #
+            # To create a google.cloud.storage.Client(), we need a
+            # project - projects in GCP are kind of like accounts in
+            # AWS, but a little more convenient to flip through and
+            # less directly tied to billing.
+            #
+            # So it turns out that `google.cloud.storage.Client()`,
+            # when reading "application default creds", requires a GCP
+            # "project" to be passed in.  Nothing you do on the
+            # command line will make it guess the right one short of
+            # (I'm not kidding) logging in again using a different
+            # method that doesn't get useful creds here, and *then*
+            # setting the project in config.
+            #
+            # So the other only option than passing this project arg
+            # in is:
+            #
+            #   # note the next --project arg is ignored for this purpose
+            #   gcloud auth application-default login --project my-project-here
+            #   gcloud auth login
+            #   # now we can set a gcloud config and have it read:
+            #   gcloud config set project my-project-here
+            #
+            if 'GCP_PROJECT' in os.environ:
+                other_args['project'] = os.environ['GCP_PROJECT']  # TODO: Mention in CONFIG.md
+            self.__default_gcs_client = google.cloud.storage.Client(credentials=gcs_creds,
+                                                                    **other_args)
```

Comment:
See https://app.asana.com/0/53283930106309/1198951309208692 for ideas on how to simplify this.